lib/spack/spack/cmd/buildcache.py

```diff
@@ -371,7 +369,9 @@ def _createtarball(env, spec_yaml=None, packages=None, add_spec=True,
             tty.debug('skipping external or virtual spec %s' %
                       match.format())
         else:
-            if add_spec:
+            lookup = spack.store.db.query_one(match)
+
+            if add_spec and lookup is not None:
                 tty.debug('adding matching spec %s' % match.format())
                 specs.add(match)
             else:
```

Comment:
The skip message should indicate whether the package was skipped intentionally or because it was not installed. There are now multiple ways to reach this `else` block